### PR TITLE
130 navigation sidebar customization

### PR DESCRIPTION
**Pull request by jejoivanic**
## Feature
A mechanism for customizing the navigation sidebar was implemented. Several components were created in order to achieve this. The usage of these components is very similar to the ones described in the issue #130.
### New components
The components created to customize the sidebar are described below:
#### Action
This component behaves as a button with a title and an action which will be triggered on the click event.
#### Link
This component behaves as a link to redirect the user to the desired path.
#### Heading
This component adds a heading to the item list which has a button to redirect to a desired path.
#### Node
This component can group and include the Action, Link and Heading components in a grouped list in the sidebar. Currently it can NOT contain another Node component in its children items.
#### Sidebar
This is the root component of the sidebar. Its possible children items can be any of the listed previously.
#### DefaultSidebar
This component will be used if none custom Sidebar is provided. Visually, it is exactly the same as it was in the Ui component, except it is implemented using the new Sidebar Components.

### Example of usage:
*CustomSidebarComponent*
```vue
...
<Sidebar>
    <Heading heading="An Example Heading" href="/!#/"/>
    <Node title="Grouped Items" icon="keyboard_arrow_up" icon-alt="keyboard_arrow_down">
      <Link title="Item_0" path="/item-0" icon="list" />
      <Link title="Item_1" path="/item-1" icon="list" />
      <Link title="Item_n" path="/item-n" icon="list" />
    </Node>
    <Action title="An action" :action="someAction" icon="accessibility" />
  </Sidebar>
...
```
*In Admin component*
```vue
...
  <Admin :authProvider="authProvider" :sidebar="CustomSidebarComponent">
...
```
### Some considerations
As the Vuetify's component v-list-group, the one that was used in the implementation, does not allow nested grouped list, our Node component can neither. In order to achieve it some inner modifications in the library component css should be done.
So the following example will not render correctly:
```vue
<Sidebar>
    <Node title="A Node" icon="keyboard_arrow_up" icon-alt="keyboard_arrow_down">
      <Node title="A Nested Node" icon="keyboard_arrow_up" icon-alt="keyboard_arrow_down">
        <Link title="An Item" path="/item" icon="list" />
      </Node>
    </Node>
    <Action title="An action" :action="someAction" icon="accessibility" />
  </Sidebar>
```

### CODE REVIEW

File: src/components/UiComponents/Sidebar/Heading.vue [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    align-center
+    :key="heading"
+  >
+    <v-flex xs6>
+      <v-subheader>
+        {{ heading }}
+      </v-subheader>
+    </v-flex>
+    <v-flex xs6 class="text-xs-center">
+      <a :href="href" class="body-2 black--text">EDIT</a>
```
[the remark this answers is not in the document]
Comment (jejoivanic) [reply src/components/UiComponents/Sidebar/Heading.vue]: I wasn't sure about this one either. For this one i justed copied the original content that was meant for the heading and parameterized the href and heading text but the *EDIT* was hardcoded.
Did you mean that, for the demo, a real-world case should be made? For example set as heading *Home* and a house icon, maybe.

File: demo/App.vue [hunk 2]
```diff
@@ -35,6 +37,8 @@ import Resource from '@components/Resource'
 import UnauthorizedView from './components/UnauthorizedView'
 import UnauthorizedCustomView from './components/UnauthorizedCustomView'
 
+import SidebarCustom from './components/SidebarCustom.vue'
```
[the remark this answers is not in the document]
Comment (jejoivanic) [reply demo/App.vue]: At first I choosed the SidebarCustom name to make it easier for the user to find. I thought "I would look for some component named Sidebar-something" but, now that you mention it, I think CustomSidebar sounds more natural and grammarly correct.

File: demo/components/SidebarCustom.vue [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<template>
+  <Sidebar>
+    <Heading heading="A Heading" href="/!#/"/>
```
[the remark this answers is not in the document]
Comment (jejoivanic) [reply demo/components/SidebarCustom.vue]: Totally agree. It sounds more like a real-world scenario.

File: src/components/UiComponents/Sidebar/Heading.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (sgobotta) [reply src/components/UiComponents/Sidebar/Heading.vue]: Yeah, I think Home would be ok. We just need text, no need to link it somewhere.

**Pull request merged**